### ISSUE TITLE
Could not deploy knowledgebase

### ISSUE THREAD

**Issue by ericchansen** [microsoft/botframework-solutions#1305]
## Project

**Name**: Virtual Assistant

**Language**: C#

## Description

Unable to deploy Azure resources. I've run this in other Azure subscriptions without issue. I'm not sure what's going wrong here. Unfortunately, the error messages aren't giving me enough to go off of.

## To Reproduce

1. Follow directions in https://github.com/Microsoft/AI/blob/master/docs/tutorials/csharp/virtualassistant.md and https://github.com/Microsoft/AI/blob/master/docs/tutorials/assistantandskilldeploymentsteps.md.

2. Manually create app at www.luis.ai and retrieve authoring key (use in command in PowerShell step).

3. Manually create application registration (use app ID in PowerShell step) and create a secret (use secret in PowerShell step).

4. Create bot using template (as described in docs). I tried this in both Visual Studio 2017 and 2019.

5. Build the bot and ensure it builds without errors.

6. Using PowerShell, attempted to run deployment script (replaced sensitive information with `xxx`).

```
.\Deployment\Scripts\deploy.ps1 -name "long-unique-name-of-my-bot" -location "westus" -luisAuthoringKey "xxx" -luisAuthoringRegion "westus" -resourceGroup "long-unique-name-of-my-bot" -appId "xxx" -appPassword "xxx"
> Creating resource group ...
> Deploying Azure services (this could take a while)...
> Updating appsettings.json ...
> Deploying cognitive models ...
> Initializing dispatch model ...
> Parsing general LU file ...
> Deploying general LUIS app ...
> Adding general app to dispatch model ...
> Parsing chitchat LU file ...
> Deploying chitchat QnA kb ...
! Could not deploy knowledgebase. Review the log for more information.
! Log: C:\xxx\VirtualAssistantSample\Deployment\Scripts\..\deploy_cognitive_models_log.txt
! Could not create knowledgebase. Skipping dispatch add.
> Parsing faq LU file ...
> Deploying faq QnA kb ...
! Could not deploy knowledgebase. Review the log for more information.
! Log: C:\xxx\VirtualAssistantSample\Deployment\Scripts\..\deploy_cognitive_models_log.txt
! Could not create knowledgebase. Skipping dispatch add.
> Creating dispatch model...
> Done.
```

Contents of `deploy_log.txt`

```
{
  "appInsights": {
    "type": "Object",
    "value": {
      "appId": "xxx",
      "instrumentationKey": "xxx"
    }
  },
  "contentModerator": {
    "type": "Object",
    "value": {
      "key": "xxx"
    }
  },
  "cosmosDb": {
    "type": "Object",
    "value": {
      "authkey": "xxx",
      "collectionId": "botstate-collection",
      "cosmosDBEndpoint": "https://xxx.documents.azure.com:443/",
      "databaseId": "botstate-db"
    }
  },
  "luis": {
    "type": "Object",
    "value": {
      "key": "xxx"
    }
  },
  "qnaMaker": {
    "type": "Object",
    "value": {
      "endpoint": "https://xxx-qnahost.azurewebsites.net",
      "key": "xxx"
    }
  },
  "storage": {
    "type": "Object",
    "value": {
      "connectionString": "DefaultEndpointsProtocol=https;AccountName=xxx;AccountKey=xxx;EndpointSuffix=core.windows.net",
      "container": "transcripts"
    }
  }
}
```

Contents of `deploy_cognitive_models_log.txt`

```
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrΓÇª           
 Confirm                            Intent Classifier   3         InProgrΓÇª           
 DirectionalReference               Entity Extractor    3         InProgrΓÇª           
 Escalate                           Intent Classifier   3         InProgrΓÇª           
 FinishTask                         Intent Classifier   3         InProgrΓÇª           
 GoBack                             Intent Classifier   3         InProgrΓÇª           
 Help                               Intent Classifier   3         InProgrΓÇª           
 Logout                             Intent Classifier   3         InProgrΓÇª           
 None                               Intent Classifier   3         InProgrΓÇª           
 ReadAloud                          Intent Classifier   3         InProgrΓÇª           
 Reject                             Intent Classifier   3         InProgrΓÇª           
 Repeat                             Intent Classifier   3         InProgrΓÇª           
 SelectAny                          Intent Classifier   3         InProgrΓÇª           
 SelectItem                         Intent Classifier   3         InProgrΓÇª           
 SelectNone                         Intent Classifier   3         InProgrΓÇª           
 ShowNext                           Intent Classifier   3         InProgrΓÇª           
 ShowPrevious                       Intent Classifier   3         InProgrΓÇª           
 StartOver                          Intent Classifier   3         InProgrΓÇª           
 Stop                               Intent Classifier   3         InProgrΓÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrΓÇª           
 Confirm                            Intent Classifier   3         InProgrΓÇª           
 DirectionalReference               Entity Extractor    3         InProgrΓÇª           
 Escalate                           Intent Classifier   3         InProgrΓÇª           
 FinishTask                         Intent Classifier   3         InProgrΓÇª           
 GoBack                             Intent Classifier   3         InProgrΓÇª           
 Help                               Intent Classifier   3         InProgrΓÇª           
 Logout                             Intent Classifier   3         InProgrΓÇª           
 None                               Intent Classifier   3         InProgrΓÇª           
 ReadAloud                          Intent Classifier   3         InProgrΓÇª           
 Reject                             Intent Classifier   3         InProgrΓÇª           
 Repeat                             Intent Classifier   3         InProgrΓÇª           
 SelectAny                          Intent Classifier   3         InProgrΓÇª           
 SelectItem                         Intent Classifier   3         InProgrΓÇª           
 SelectNone                         Intent Classifier   3         InProgrΓÇª           
 ShowNext                           Intent Classifier   3         InProgrΓÇª           
 ShowPrevious                       Intent Classifier   3         InProgrΓÇª           
 StartOver                          Intent Classifier   3         InProgrΓÇª           
 Stop                               Intent Classifier   3         InProgrΓÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrΓÇª           
 Confirm                            Intent Classifier   3         InProgrΓÇª           
 DirectionalReference               Entity Extractor    3         InProgrΓÇª           
 Escalate                           Intent Classifier   3         InProgrΓÇª           
 FinishTask                         Intent Classifier   3         InProgrΓÇª           
 GoBack                             Intent Classifier   3         InProgrΓÇª           
 Help                               Intent Classifier   3         InProgrΓÇª           
 Logout                             Intent Classifier   3         InProgrΓÇª           
 None                               Intent Classifier   3         InProgrΓÇª           
 ReadAloud                          Intent Classifier   3         InProgrΓÇª           
 Reject                             Intent Classifier   3         InProgrΓÇª           
 Repeat                             Intent Classifier   3         InProgrΓÇª           
 SelectAny                          Intent Classifier   3         InProgrΓÇª           
 SelectItem                         Intent Classifier   3         InProgrΓÇª           
 SelectNone                         Intent Classifier   3         InProgrΓÇª           
 ShowNext                           Intent Classifier   3         InProgrΓÇª           
 ShowPrevious                       Intent Classifier   3         InProgrΓÇª           
 StartOver                          Intent Classifier   3         InProgrΓÇª           
 Stop                               Intent Classifier   3         InProgrΓÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrΓÇª           
 Confirm                            Intent Classifier   3         InProgrΓÇª           
 DirectionalReference               Entity Extractor    3         InProgrΓÇª           
 Escalate                           Intent Classifier   3         InProgrΓÇª           
 FinishTask                         Intent Classifier   3         InProgrΓÇª           
 GoBack                             Intent Classifier   3         InProgrΓÇª           
 Help                               Intent Classifier   3         InProgrΓÇª           
 Logout                             Intent Classifier   3         InProgrΓÇª           
 None                               Intent Classifier   3         InProgrΓÇª           
 ReadAloud                          Intent Classifier   3         InProgrΓÇª           
 Reject                             Intent Classifier   3         InProgrΓÇª           
 Repeat                             Intent Classifier   3         InProgrΓÇª           
 SelectAny                          Intent Classifier   3         InProgrΓÇª           
 SelectItem                         Intent Classifier   3         InProgrΓÇª           
 SelectNone                         Intent Classifier   3         InProgrΓÇª           
 ShowNext                           Intent Classifier   3         InProgrΓÇª           
 ShowPrevious                       Intent Classifier   3         InProgrΓÇª           
 StartOver                          Intent Classifier   3         InProgrΓÇª           
 Stop                               Intent Classifier   3         InProgrΓÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrΓÇª           
 Confirm                            Intent Classifier   3         InProgrΓÇª           
 DirectionalReference               Entity Extractor    3         InProgrΓÇª           
 Escalate                           Intent Classifier   3         InProgrΓÇª           
 FinishTask                         Intent Classifier   3         InProgrΓÇª           
 GoBack                             Intent Classifier   3         InProgrΓÇª           
 Help                               Intent Classifier   3         InProgrΓÇª           
 Logout                             Intent Classifier   3         InProgrΓÇª           
 None                               Intent Classifier   3         InProgrΓÇª           
 ReadAloud                          Intent Classifier   3         InProgrΓÇª           
 Reject                             Intent Classifier   3         InProgrΓÇª           
 Repeat                             Intent Classifier   3         InProgrΓÇª           
 SelectAny                          Intent Classifier   3         InProgrΓÇª           
 SelectItem                         Intent Classifier   3         InProgrΓÇª           
 SelectNone                         Intent Classifier   3         InProgrΓÇª           
 ShowNext                           Intent Classifier   3         InProgrΓÇª           
 ShowPrevious                       Intent Classifier   3         InProgrΓÇª           
 StartOver                          Intent Classifier   3         InProgrΓÇª           
 Stop                               Intent Classifier   3         InProgrΓÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrΓÇª           
 Confirm                            Intent Classifier   3         InProgrΓÇª           
 DirectionalReference               Entity Extractor    3         InProgrΓÇª           
 Escalate                           Intent Classifier   3         InProgrΓÇª           
 FinishTask                         Intent Classifier   3         InProgrΓÇª           
 GoBack                             Intent Classifier   3         InProgrΓÇª           
 Help                               Intent Classifier   3         InProgrΓÇª           
 Logout                             Intent Classifier   3         InProgrΓÇª           
 None                               Intent Classifier   3         InProgrΓÇª           
 ReadAloud                          Intent Classifier   3         InProgrΓÇª           
 Reject                             Intent Classifier   3         InProgrΓÇª           
 Repeat                             Intent Classifier   3         InProgrΓÇª           
 SelectAny                          Intent Classifier   3         InProgrΓÇª           
 SelectItem                         Intent Classifier   3         InProgrΓÇª           
 SelectNone                         Intent Classifier   3         InProgrΓÇª           
 ShowNext                           Intent Classifier   3         InProgrΓÇª           
 ShowPrevious                       Intent Classifier   3         InProgrΓÇª           
 StartOver                          Intent Classifier   3         InProgrΓÇª           
 Stop                               Intent Classifier   3         InProgrΓÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrΓÇª           
 Confirm                            Intent Classifier   3         InProgrΓÇª           
 DirectionalReference               Entity Extractor    3         InProgrΓÇª           
 Escalate                           Intent Classifier   3         InProgrΓÇª           
 FinishTask                         Intent Classifier   3         InProgrΓÇª           
 GoBack                             Intent Classifier   3         InProgrΓÇª           
 Help                               Intent Classifier   3         InProgrΓÇª           
 Logout                             Intent Classifier   3         InProgrΓÇª           
 None                               Intent Classifier   3         InProgrΓÇª           
 ReadAloud                          Intent Classifier   3         InProgrΓÇª           
 Reject                             Intent Classifier   3         InProgrΓÇª           
 Repeat                             Intent Classifier   3         InProgrΓÇª           
 SelectAny                          Intent Classifier   3         InProgrΓÇª           
 SelectItem                         Intent Classifier   3         InProgrΓÇª           
 SelectNone                         Intent Classifier   3         InProgrΓÇª           
 ShowNext                           Intent Classifier   3         InProgrΓÇª           
 ShowPrevious                       Intent Classifier   3         InProgrΓÇª           
 StartOver                          Intent Classifier   3         InProgrΓÇª           
 Stop                               Intent Classifier   3         InProgrΓÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrΓÇª           
 Confirm                            Intent Classifier   3         InProgrΓÇª           
 DirectionalReference               Entity Extractor    3         InProgrΓÇª           
 Escalate                           Intent Classifier   3         InProgrΓÇª           
 FinishTask                         Intent Classifier   3         InProgrΓÇª           
 GoBack                             Intent Classifier   3         InProgrΓÇª           
 Help                               Intent Classifier   3         InProgrΓÇª           
 Logout                             Intent Classifier   3         InProgrΓÇª           
 None                               Intent Classifier   3         InProgrΓÇª           
 ReadAloud                          Intent Classifier   3         InProgrΓÇª           
 Reject                             Intent Classifier   3         InProgrΓÇª           
 Repeat                             Intent Classifier   3         InProgrΓÇª           
 SelectAny                          Intent Classifier   3         InProgrΓÇª           
 SelectItem                         Intent Classifier   3         InProgrΓÇª           
 SelectNone                         Intent Classifier   3         InProgrΓÇª           
 ShowNext                           Intent Classifier   3         InProgrΓÇª           
 ShowPrevious                       Intent Classifier   3         InProgrΓÇª           
 StartOver                          Intent Classifier   3         InProgrΓÇª           
 Stop                               Intent Classifier   3         InProgrΓÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrΓÇª           
 Confirm                            Intent Classifier   3         InProgrΓÇª           
 DirectionalReference               Entity Extractor    3         InProgrΓÇª           
 Escalate                           Intent Classifier   3         InProgrΓÇª           
 FinishTask                         Intent Classifier   3         InProgrΓÇª           
 GoBack                             Intent Classifier   3         InProgrΓÇª           
 Help                               Intent Classifier   3         InProgrΓÇª           
 Logout                             Intent Classifier   3         InProgrΓÇª           
 None                               Intent Classifier   3         InProgrΓÇª           
 ReadAloud                          Intent Classifier   3         InProgrΓÇª           
 Reject                             Intent Classifier   3         InProgrΓÇª           
 Repeat                             Intent Classifier   3         InProgrΓÇª           
 SelectAny                          Intent Classifier   3         InProgrΓÇª           
 SelectItem                         Intent Classifier   3         InProgrΓÇª           
 SelectNone                         Intent Classifier   3         InProgrΓÇª           
 ShowNext                           Intent Classifier   3         InProgrΓÇª           
 ShowPrevious                       Intent Classifier   3         InProgrΓÇª           
 StartOver                          Intent Classifier   3         InProgrΓÇª           
 Stop                               Intent Classifier   3         InProgrΓÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   3         InProgrΓÇª           
 Confirm                            Intent Classifier   3         InProgrΓÇª           
 DirectionalReference               Entity Extractor    3         InProgrΓÇª           
 Escalate                           Intent Classifier   3         InProgrΓÇª           
 FinishTask                         Intent Classifier   3         InProgrΓÇª           
 GoBack                             Intent Classifier   3         InProgrΓÇª           
 Help                               Intent Classifier   3         InProgrΓÇª           
 Logout                             Intent Classifier   3         InProgrΓÇª           
 None                               Intent Classifier   3         InProgrΓÇª           
 ReadAloud                          Intent Classifier   3         InProgrΓÇª           
 Reject                             Intent Classifier   3         InProgrΓÇª           
 Repeat                             Intent Classifier   3         InProgrΓÇª           
 SelectAny                          Intent Classifier   3         InProgrΓÇª           
 SelectItem                         Intent Classifier   3         InProgrΓÇª           
 SelectNone                         Intent Classifier   3         InProgrΓÇª           
 ShowNext                           Intent Classifier   3         InProgrΓÇª           
 ShowPrevious                       Intent Classifier   3         InProgrΓÇª           
 StartOver                          Intent Classifier   3         InProgrΓÇª           
 Stop                               Intent Classifier   3         InProgrΓÇª           
[31m Model                             [39m[31m Type               [39m[31m StatusId [39m[31m Status   [39m[31m          [39m
 Cancel                             Intent Classifier   0         Success            
 Confirm                            Intent Classifier   0         Success            
 Escalate                           Intent Classifier   0         Success            
 FinishTask                         Intent Classifier   0         Success            
 GoBack                             Intent Classifier   0         Success            
 Help                               Intent Classifier   0         Success            
 Logout                             Intent Classifier   0         Success            
 None                               Intent Classifier   0         Success            
 ReadAloud                          Intent Classifier   0         Success            
 Reject                             Intent Classifier   0         Success            
 Repeat                             Intent Classifier   0         Success            
 SelectAny                          Intent Classifier   0         Success            
 SelectItem                         Intent Classifier   0         Success            
 SelectNone                         Intent Classifier   0         Success            
 ShowNext                           Intent Classifier   0         Success            
 ShowPrevious                       Intent Classifier   0         Success            
 StartOver                          Intent Classifier   0         Success            
 Stop                               Intent Classifier   0         Success            
 DirectionalReference               Entity Extractor    0         Success            
'
NotStarted.              '
Failed..              
Error: {
    "operationState": "Failed",
    "createdTimestamp": "2019-05-07T02:44:15Z",
    "lastActionTimestamp": "2019-05-07T02:44:17Z",
    "userId": "83362c116f6a4b5394dc364e775a3729",
    "operationId": "01d8744c-4287-4cd0-a838-08be08d57f28"
}

QnA Maker Command line interface - ┬⌐ 2018 Microsoft Corporation

Create a new knowledgebase

Usage:
qnamaker create kb --in createKbPayload.json --name <kbname> [--wait]

Command arguments are:
 --in CreateKbDTO.json  The CreateKbDTO object to send in the body of the request                                                                                                                                   
                        {                                                                                                                                                                                           
                          "name": "QnA Maker FAQ",                                                                                                                                                                  
                          "qnaList": [                                                                                                                                                                              
                            {                                                                                                                                                                                       
                              "id": 0,                                                                                                                                                                              
                              "answer": "You can change the default message if you use the QnAMakerDialog. See this for details: https://docs.botframework.com/en-us/azure-bot-service/templates/qnamaker/#navtitlΓÇª 
                              "source": "Custom Editorial",                                                                                                                                                         
                              "questions": [                                                                                                                                                                        
                                "How can I change the default message from QnA Maker?"                                                                                                                              
                              ],                                                                                                                                                                                    
                              "metadata": []                                                                                                                                                                        
                            },                                                                                                                                                                                      
                            {                                                                                                                                                                                       
                              "id": 0,                                                                                                                                                                              
                              "answer": "You can use our REST apis to manage your KB. See here for details:                                                                                                         
                        https://westus.dev.cognitive.microsoft.com/docs/services/58994a073d9e04097c7ba6fe/operations/58994a073d9e041ad42d9baa",                                                                     
                              "source": "Custom Editorial",                                                                                                                                                         
                              "questions": [                                                                                                                                                                        
                                "How do I programmatically update my KB?"                                                                                                                                           
                              ],                                                                                                                                                                                    
                              "metadata": [                                                                                                                                                                         
                                {                                                                                                                                                                                   
                                  "name": "category",                                                                                                                                                               
                                  "value": "api"                                                                                                                                                                    
                                }                                                                                                                                                                                   
                              ]                                                                                                                                                                                     
                            }                                                                                                                                                                                       
                          ],                                                                                                                                                                                        
                          "urls": [                                                                                                                                                                                 
                            "https://docs.microsoft.com/en-in/azure/cognitive-services/qnamaker/faqs",                                                                                                              
                            "https://docs.microsoft.com/en-us/bot-framework/resources-bot-framework-faq"                                                                                                            
                          ],                                                                                                                                                                                        
                          "files": [                                                                                                                                                                                
                            {                                                                                                                                                                                       
                              "fileName": "SurfaceManual.pdf",                                                                                                                                                      
                              "fileUri": "https://download.microsoft.com/download/2/9/B/29B20383-302C-4517-A006-B0186F04BE28/surface-pro-4-user-guide-EN.pdf"                                                       
                            }                                                                                                                                                                                       
                          ]                                                                                                                                                                                         
                        }                                                                                                                                                                                           
 --name <string>        name of the kb you want to create                                                                                                                                                           
 --wait <boolean>       Wait for the operation to complete                                                                                                                                                          
 --msbot                (OPTIONAL) Format the output as json for piping into msbot connect qna command                                                                                                              

Configuration and Overrides:
 --subscriptionKey      Specifies the qnamaker subscription key/access keys (found on the Cognitive Services Azure portal page under "access keys"). Overrides the .qnamakerrc value and the                        
 <key>                  QNAMAKER_SUBSCRIPTION_KEY environment variable.                                                                                                                                             
 --hostname <url>       Specifies the url for your private QnA service. Overrides the .qnamakerrc value and the QNAMAKER_HOSTNAME environment variable.                                                             
 --endpointKey <key>    Specifies the endpoint key for your private QnA service.(from qnamaker.ai portal user settings page). Overrides the .qnamakerrc value and the QNAMAKER_ENDPOINTKEY environment variable.    
 --kbId <kbId>          Specifies the active qnamaker knowledgebase id. Overrides the .qnamakerrc value and the QNAMAKER_KBID environment variable.                                                                 
 --stdin                Specifies qnamaker configuration is being passed via stdin. (Example: output of msbot get service) Overrides the .qnamakerrc value and the QNAMAKER_KBID environment variable.              

Global Arguments:
 --help,    -h          Prints this help file.                                                                                                                                                                      
 --version, -v          Prints the version of this cli tool                                                                                                                                                         
 --force,   -f          Do not prompt for confirmation, force the operation                                                                                                                                         
 --!                    Dumps all documented commands to the console with descriptions                                                                                                                              

'
NotStarted.              '
Failed..              
Error: {
    "operationState": "Failed",
    "createdTimestamp": "2019-05-07T02:44:20Z",
    "lastActionTimestamp": "2019-05-07T02:44:21Z",
    "userId": "83362c116f6a4b5394dc364e775a3729",
    "operationId": "7a96295c-b254-4b85-a233-9a63365bd320"
}

QnA Maker Command line interface - ┬⌐ 2018 Microsoft Corporation

Create a new knowledgebase

Usage:
qnamaker create kb --in createKbPayload.json --name <kbname> [--wait]

Command arguments are:
 --in CreateKbDTO.json  The CreateKbDTO object to send in the body of the request                                                                                                                                   
                        {                                                                                                                                                                                           
                          "name": "QnA Maker FAQ",                                                                                                                                                                  
                          "qnaList": [                                                                                                                                                                              
                            {                                                                                                                                                                                       
                              "id": 0,                                                                                                                                                                              
                              "answer": "You can change the default message if you use the QnAMakerDialog. See this for details: https://docs.botframework.com/en-us/azure-bot-service/templates/qnamaker/#navtitlΓÇª 
                              "source": "Custom Editorial",                                                                                                                                                         
                              "questions": [                                                                                                                                                                        
                                "How can I change the default message from QnA Maker?"                                                                                                                              
                              ],                                                                                                                                                                                    
                              "metadata": []                                                                                                                                                                        
                            },                                                                                                                                                                                      
                            {                                                                                                                                                                                       
                              "id": 0,                                                                                                                                                                              
                              "answer": "You can use our REST apis to manage your KB. See here for details:                                                                                                         
                        https://westus.dev.cognitive.microsoft.com/docs/services/58994a073d9e04097c7ba6fe/operations/58994a073d9e041ad42d9baa",                                                                     
                              "source": "Custom Editorial",                                                                                                                                                         
                              "questions": [                                                                                                                                                                        
                                "How do I programmatically update my KB?"                                                                                                                                           
                              ],                                                                                                                                                                                    
                              "metadata": [                                                                                                                                                                         
                                {                                                                                                                                                                                   
                                  "name": "category",                                                                                                                                                               
                                  "value": "api"                                                                                                                                                                    
                                }                                                                                                                                                                                   
                              ]                                                                                                                                                                                     
                            }                                                                                                                                                                                       
                          ],                                                                                                                                                                                        
                          "urls": [                                                                                                                                                                                 
                            "https://docs.microsoft.com/en-in/azure/cognitive-services/qnamaker/faqs",                                                                                                              
                            "https://docs.microsoft.com/en-us/bot-framework/resources-bot-framework-faq"                                                                                                            
                          ],                                                                                                                                                                                        
                          "files": [                                                                                                                                                                                
                            {                                                                                                                                                                                       
                              "fileName": "SurfaceManual.pdf",                                                                                                                                                      
                              "fileUri": "https://download.microsoft.com/download/2/9/B/29B20383-302C-4517-A006-B0186F04BE28/surface-pro-4-user-guide-EN.pdf"                                                       
                            }                                                                                                                                                                                       
                          ]                                                                                                                                                                                         
                        }                                                                                                                                                                                           
 --name <string>        name of the kb you want to create                                                                                                                                                           
 --wait <boolean>       Wait for the operation to complete                                                                                                                                                          
 --msbot                (OPTIONAL) Format the output as json for piping into msbot connect qna command                                                                                                              

Configuration and Overrides:
 --subscriptionKey      Specifies the qnamaker subscription key/access keys (found on the Cognitive Services Azure portal page under "access keys"). Overrides the .qnamakerrc value and the                        
 <key>                  QNAMAKER_SUBSCRIPTION_KEY environment variable.                                                                                                                                             
 --hostname <url>       Specifies the url for your private QnA service. Overrides the .qnamakerrc value and the QNAMAKER_HOSTNAME environment variable.                                                             
 --endpointKey <key>    Specifies the endpoint key for your private QnA service.(from qnamaker.ai portal user settings page). Overrides the .qnamakerrc value and the QNAMAKER_ENDPOINTKEY environment variable.    
 --kbId <kbId>          Specifies the active qnamaker knowledgebase id. Overrides the .qnamakerrc value and the QNAMAKER_KBID environment variable.                                                                 
 --stdin                Specifies qnamaker configuration is being passed via stdin. (Example: output of msbot get service) Overrides the .qnamakerrc value and the QNAMAKER_KBID environment variable.              

Global Arguments:
 --help,    -h          Prints this help file.                                                                                                                                                                      
 --version, -v          Prints the version of this cli tool                                                                                                                                                         
 --force,   -f          Do not prompt for confirmation, force the operation                                                                                                                                         
 --!                    Dumps all documented commands to the console with descriptions                                                                                                                              

Exporting services for dispatch...
Exporting xxx-virtual-agenten_general LUIS application...
Creating dispatch LUIS model json...
Creating training data...
WARNING: very few samples (breaking-point for debugging purpose), foldSize=0.6, foldsCount=5, Samples.Length=3, category=None, TrainingSamplesCount=2, TestingSamplesCount=1
Updating xxx-virtual-agenten_Dispatch model...
Importing xxx-virtual-agenten_Dispatch model...
Training xxx-virtual-agenten_Dispatch model...
Setting up intents to child services mapping for xxx-virtual-agenten_Dispatch...
Publishing xxx-virtual-agenten_Dispatch model...
Training xxx-virtual-agenten_general...
Publishing xxx-virtual-agenten_general...
Writing summary file ('test_prediction')...
Please review your dispatch model in C:\xxx\VirtualAssistantSample\Deployment\Resources\Dispatch\en\Summary.html
```

## Expected behavior

I've run this with other Azure subscriptions and everything works as expected.

## Screenshots

None.

## Additional context

None.

**Comment by darrenj** [comment]
Thanks @ericchansen and for the detailed repro. We noticed the same earlier and have encountered (a new?!) race condition with QnAMaker reporting that it's provisioned resources but actually isn't ready hence publishing fails. This was working until the weekend but as with race conditions things can throw it out.

We've just had word back from the QnAMaker team that we have to implement artificial delays :-(  We've just implemented a change (PR #1306) which looks good and I've just tested. We'll push new VSIX tomorrow for new projects.

If you have time please try and pull the qna functions ps script from [here](https://github.com/microsoft/AI/pull/1306) and let us know. Thanks again!

**Comment by darrenj** [comment]
Update - PR looks good.

![image](https://user-images.githubusercontent.com/33519520/57275469-8061cb80-7053-11e9-9e4d-86487d3f7b6d.png)
